airbyte-db/lib/src/main/java/io/airbyte/db/jdbc/streaming/AdaptiveStreamingQueryConfig.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.db.jdbc.streaming;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AdaptiveStreamingQueryConfig implements JdbcStreamingQueryConfig {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AdaptiveStreamingQueryConfig.class);
+
+  private final FetchSizeEstimator fetchSizeEstimator;
+  private int currentFetchSize;
+
+  public AdaptiveStreamingQueryConfig() {
+    this.fetchSizeEstimator = TwoStageSizeEstimator.getInstance();
+    this.currentFetchSize = FetchSizeConstants.INITIAL_SAMPLE_SIZE;
+  }
+
+  @Override
+  public void initialize(final Connection connection, final Statement preparedStatement) throws SQLException {
+    connection.setAutoCommit(false);
+    preparedStatement.setFetchSize(FetchSizeConstants.INITIAL_SAMPLE_SIZE);
+    currentFetchSize = FetchSizeConstants.INITIAL_SAMPLE_SIZE;
+    LOGGER.info("Set initial fetch size: {}", preparedStatement.getFetchSize());
```

Comment:
the log should probably say what the size unit is, e.g. is it number of rows or bytes?

Reply:
Good point. Will do.